profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -31,54 +32,92 @@ export function arrayShallowCopying() {
 }
 
 export function UseEffectThrashing({ fetchURL, label }) {
+  const [{ data, error, loading }, setQuery] = useState({
+    data: undefined,
+    error: undefined,
+    loading: false,
+  });
+
+  // assuming we want to fetch on mount, the button is misleading
```

Comment:
the button makes me think maybe we want to fetch on click, as for the effect thrashing, the user of the component should understand that changing the fetchURL prop would cause a new fetch, otherwise we can change it to be on mount but data could be stale